counsel.el

```diff
@@ -2699,6 +2719,34 @@ Note: don't use single quotes for the regex."
 (ivy-set-occur 'counsel-rg 'counsel-ag-occur)
 (ivy-set-display-transformer 'counsel-rg 'counsel-git-grep-transformer)
 
+(defcustom counsel-rg-pcre-flag 'auto
+  "PCRE flag used by Ripgrep.
+
+If `auto', flag will be determined automatically.
+
+If `nil', PCRE support is disabled unless \"--pcre2\" or \"-P\"
+is part of `counsel-rg-base-command'.
+
+If a string, pass value to Ripgrep when PCRE support is
+requested.
+
+PCRE is required to support multi-pass regex builders such as
+`ivy--regex-plus' and `ivy--regex-ignore-order'.
+
+Note: only Ripgrep version 0.10.0 and later support PCRE."
+  :type '(choice (const :tag "Auto detect" auto)
+                 (const :tag "Disable" nil)
+                 (string :tag "PCRE flag")))
+
+(defun counsel--rg-version ()
+  "Return Ripgrep version string or \"0.0.0\" on failure."
+  (string-match "\\(?:\\`\\| \\)\\([^ =]+\\) " counsel-rg-base-command)
+  (let* ((rg (match-string 1 counsel-rg-base-command))
```

Comment:
We should assume neither that the first `string-match` is going to succeed, nor that `version` will be non-`nil`:
```el
(let ((version (and (string-match "\\(?:\\`\\| \\)\\([^ =]+\\) "
                                  counsel-rg-base-command)
                    (ignore-errors
                      (counsel--call (match-string 1 counsel-rg-base-command)
                                     "--version")))))
  (if (and version (string-match "\\`ripgrep \\([.0-9]+\\)" version))
      (match-string 1 version)
    "0.0.0"))
```